/* Copyright (c) 2012 Stanford University
 *
 * Permission to use, copy, modify, and distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */

syntax = "proto2";

package RAMCloud.ProtoBuf;

/// Serialization of a Histogram object. Please see Histogram.h's member
/// variable documentation for details on the fields below.
message Histogram {
    required fixed64 num_buckets = 1;
    required fixed64 bucket_width = 2;

    // Each pair stores a unique bucket index and the count. Unclear if doing
    // this an omitting empty buckets will be a net win over also including
    // the index value.
    message Bucket {
        required fixed64 index = 1;
        required fixed64 count = 2;
    }
    repeated Bucket bucket = 3;

    required fixed64 sample_sum_high = 4;
    required fixed64 sample_sum_low = 5;
    required fixed64 outliers = 6;
    required fixed64 max = 7;
    required fixed64 min = 8;
}
